syntax = "proto3";

package deep_oj;

option go_package = "github.com/d1guo/deep_oj/pkg/proto";

// -------------------- 枚举定义 --------------------

enum Language {
    LANGUAGE_UNSPECIFIED = 0;
    CPP = 1;
    JAVA = 2;
    PYTHON = 3;
    GO = 4;
}

enum JudgeStatus {
    STATUS_UNSPECIFIED = 0;
    ACCEPTED = 1;              // AC
    WRONG_ANSWER = 2;          // WA
    COMPILE_ERROR = 3;         // CE
    TIME_LIMIT_EXCEEDED = 4;   // TLE
    MEMORY_LIMIT_EXCEEDED = 5; // MLE
    RUNTIME_ERROR = 6;         // RE
    SYSTEM_ERROR = 7;          // SE
}

// -------------------- 消息体定义 --------------------

// 任务请求（由 Worker Stream 消费链路反序列化）
message TaskRequest {
    // 必须匹配 ^[A-Za-z0-9_-]{1,64}$，禁止路径分隔符。
    string job_id = 1;
    
    // 【关键修改】这里改成 bytes！
    // 避免 string 导致的自动转义 (backslash bug)
    // 接收端收到后：std::string code(req.code().begin(), req.code().end());
    bytes code = 2; 
    
    Language language = 3;
    int32 time_limit = 4;      // ms
    int32 memory_limit = 5;    // KB
    
    string cache_key = 6;
    int64 submit_time = 7; // Unix timestamp (ms) for timeout detection
    uint32 problem_id = 8;
    string trace_id = 9;   // [New] Full-link Trace ID
}

// 空响应 (占位符)
message TaskResponse {
    string message = 1; // [Fix] Add message field
}
